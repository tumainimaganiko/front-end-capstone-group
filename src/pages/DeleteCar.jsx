/* eslint-disable no-alert */
import React, { useEffect } from 'react';
import propTypes from 'prop-types';
import { useDispatch, useSelector } from 'react-redux';
import { deleteCar, fetchCars } from '../redux/cars/carsSlice';
import Loader from '../components/loader/Loader';

const DeleteCar = ({ car }) => {
  const dispatch = useDispatch();

  const handleDelete = () => {
    if (window.confirm('Are you sure you want to delete this car?')) {
      dispatch(deleteCar(car.id));
    }
  };

  return (
    <section className="text-gray-600 body-font pt-4">
      <div className="bg-gray-200 bg-opacity-50 p-2 rounded-lg shadow-lg flex flex-row sm:flex-col items-center justify-between text-center">
        <img src={car.image_url} alt={car.name} className="sm:w-40 w-16 rounded-full mb-2" />
        <p className="sm:text-xl text-sm font-semibold mb-2 mx-3">{car.name}</p>
        <div className="flex items-center sm:space-x-4">
          <button
            type="button"
            onClick={handleDelete}
            className="bg-red-500 hover:bg-red-600 text-white text-sm p-1 sm:py-2 sm:px-4 rounded-md transition duration-300 ease-in-out"
          >
            Delete
          </button>
        </div>
      </div>
    </section>
  );
};

const CarList = () => {
  const dispatch = useDispatch();
  const cars = useSelector((state) => state.cars.cars);
  const isLoading = useSelector((state) => state.cars.isLoading);

  useEffect(() => {
    dispatch(fetchCars());
  }, [dispatch]);

  if (isLoading) {
    return <Loader />;
  }

  return (
    <div className="min-h-screen bg-lime-300">
      {cars.map((car) => (
        <DeleteCar key={car.id} car={car} />
      ))}
    </div>
  );
};

export default CarList;

DeleteCar.propTypes = {
  car: propTypes.shape({
    id: propTypes.number,
    name: propTypes.string,
    details: propTypes.string,
    image_url: propTypes.string,
  }).isRequired,
};
