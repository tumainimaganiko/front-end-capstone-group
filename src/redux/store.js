import { configureStore } from '@reduxjs/toolkit';
import reservationReducer from './reservations/reservationSlice';
import carsReducer from './cars/carsSlice';
import modelsReducer from './cars/modelsSlice';

const store = configureStore({
  reducer: {
    reservations: reservationReducer,
    cars: carsReducer,
    models: modelsReducer,
  },
});

export default store;
