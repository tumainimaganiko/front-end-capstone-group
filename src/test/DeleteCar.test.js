import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/store';
import DeleteCar from '../pages/DeleteCar';

it('Render a DeleteCar', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <MemoryRouter>
          <DeleteCar />
        </MemoryRouter>
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
