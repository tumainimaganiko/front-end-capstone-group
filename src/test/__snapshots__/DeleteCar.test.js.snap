// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render a DeleteCar 1`] = `
<div
  className="min-h-screen bg-lime-300"
/>
`;
